api/src/paths/project/{projectId}/survey/{surveyId}/critters/{critterId}/deployments/{bctwDeploymentId}.ts

```diff
@@ -0,0 +1,114 @@
+import { getDBConnection } from '../../../../../../../../database/db';
+import { authorizeRequestHandler } from '../../../../../../../../request-handlers/security/authorization';
+import { BctwService } from '../../../../../../../../services/bctw-service';
+import { ICritterbaseUser } from '../../../../../../../../services/critterbase-service';
+import { SurveyCritterService } from '../../../../../../../../services/survey-critter-service';
+import { getLogger } from '../../../../../../../../utils/logger';
+
+const defaultLog = getLogger(
+  'paths/project/{projectId}/survey/{surveyId}/critters/{critterId}/deployments/{bctwDeploymentId}'
+);
+export const DELETE: Operation = [
+  authorizeRequestHandler((req) => {
+    return {
+      or: [
+        {
+          validProjectPermissions: [PROJECT_PERMISSION.COORDINATOR, PROJECT_PERMISSION.COLLABORATOR],
+          projectId: Number(req.params.projectId),
+          discriminator: 'ProjectPermission'
+        },
+        {
+          validSystemRoles: [SYSTEM_ROLE.DATA_ADMINISTRATOR],
+          discriminator: 'SystemRole'
+        }
+      ]
+    };
+  }),
+  deleteDeployment()
+];
+
+DELETE.apiDoc = {
+  description: 'Deletes the deployment record in SIMS, and soft deletes the record in BCTW.',
+  tags: ['bctw'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  parameters: [
+    {
+      in: 'path',
+      name: 'surveyId',
+      schema: {
+        type: 'number'
+      },
+      required: true
+    },
+    {
+      in: 'path',
+      name: 'critterId',
+      schema: {
+        type: 'number'
+      }
+    },
+    {
+      in: 'path',
+      name: 'bctwDeploymentId',
+      schema: {
+        type: 'string',
+        format: 'uuid'
+      }
+    }
+  ],
+  responses: {
+    200: {
+      description: 'Remove deployment successfully.'
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+export function deleteDeployment(): RequestHandler {
+  return async (req, res) => {
+    const user: ICritterbaseUser = {
+      keycloak_guid: req['system_user']?.user_guid,
+      username: req['system_user']?.user_identifier
+    };
+    const deploymentId = String(req.params.bctwDeploymentId);
+    const critterId = Number(req.params.critterId);
+    const connection = getDBConnection(req['keycloak_token']);
+    const surveyCritterService = new SurveyCritterService(connection);
+    const bctw = new BctwService(user);
+    try {
+      await connection.open();
+      const surveyEntry = await surveyCritterService.removeDeployment(critterId, deploymentId);
+      await bctw.deleteDeployment(deploymentId);
+      await connection.commit();
+      return res.status(200).json(surveyEntry);
```

Comment:
Survey entry is undefined because of the void return in the service. Maybe its worth just dropping the .json in the response all together. We should also probably chose a consistent way of doing this between this delete and the update endpoint which both have void returns.

Reply:
Updated to be in line with those other responses.